subxt/src/client/online_client.rs

```diff
@@ -209,34 +227,120 @@ impl<T: Config> ClientRuntimeUpdater<T> {
         &curr.runtime_version != new
     }
 
+    fn do_update(&self, update: Update) {
+        let mut writable = self.0.inner.write();
+        writable.metadata = update.metadata;
+        writable.runtime_version = update.runtime_version;
+    }
+
+    /// Tries to apply a new update.
+    ///
+    /// Returns `Ok(UpgradeResult)` which indicates whether the upgrade was successful or not
+    /// but it is not treated as hard error because it may appear during normal use.
+    ///
+    /// Returns `Err(Error)` if the metadata couldn't be fetched.
+    pub async fn apply_update(&self, update: Update) -> Result<UpgradeResult, Error> {
+        if !self.is_runtime_version_different(&update.runtime_version) {
+            return Ok(UpgradeResult::SameVersion)
+        }
+
+        let metadata = self.0.rpc.metadata().await?;
+
+        if update.metadata.runtime_metadata() != metadata.runtime_metadata() {
```

Comment:
not sure if this is correct...

Reply:
It's quite an expensive check to do relatively speaking. Anyway, if the RuntimeVersion is different then we need to call `do_update` regardless to set that new RuntimeVersion regardless.

Maybe just have `SameVersion` and `Success` and ignore this possibility?

Reply:
ah, I think it's fine to remove that check as the subscription items should anyway come in order so we would apply the updates in the same order. Got it, your comment was good misunderstood that